pkg/stack/stack_processor_utils.go

```diff
@@ -235,6 +236,22 @@ func GetGlobMatches(pattern string) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// There seems to be some underlying issues with `doublestar.Glob` not returning any matches (and no error) in concurrent execution
+	// Check if `doublestar.Glob` returned any matches.
+	// If not, try it again
+	if matches == nil {
+		time.Sleep(1 * time.Second)
+		matches, err = doublestar.Glob(pattern)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
https://github.com/bmatcuk/doublestar/blob/34e84fce94f2653a09f9660de28b7af951fcb40d/glob.go#L31-L34

> If the pattern has a `**` anywhere but the very end, GlobWalk is more
> performant because it can get away with less allocations. If the pattern
> ends in a `**`, both methods are pretty much the same, but Glob has a
>  _very_ slight advantage because of lower function call overhead.

Have we looked into `GlobWalk`? Not that it seems to mention this issue, but maybe it functions differently and is more suitable for concurrent execution

Reply:
using new version of `doublestar` now (thanks to @Nuru), hopefully it will work better